runtime/vm/vmthread.cpp

```diff
@@ -292,7 +292,13 @@ allocateVMThread(J9JavaVM * vm, omrthread_t osThread, UDATA privateFlags, void *
 	}
 #if defined(J9VM_OPT_CRIU_SUPPORT)
 	if (VM_CRIUHelpers::isJVMInSingleThreadMode(vm) && VM_VMHelpers::threadCanRunJavaCode(newThread)) {
-		setHaltFlag(newThread, J9_PUBLIC_FLAGS_HALT_THREAD_FOR_CHECKPOINT);
+		/* If JDWP is enabled, new JDWP debug threads should not be halted. */
+		if (!vm->checkpointState.isJdwpEnabled
+			|| (NULL == threadName)
+			|| (0 != strncmp("JDWP", threadName, 4))
```

Comment:
This seems rather fragile; what prevents users from being affected if their thread names start with "JDWP"?

Reply:
Added additional checks for if the thread is a daemon and part of the `system` thread group as well. This should help eliminate false positives.